File: backend/schemas/statistics.py
"""
Statistics Schemas
"""
from pydantic import BaseModel
from typing import List, Dict


class PaymentDistribution(BaseModel):
    """支付方式分布"""

    count: int
    amount: float


class MonthlyTrend(BaseModel):
    """月度趋势"""

    month: str
    amount: float
    count: int


class TopProduct(BaseModel):
    """Top产品"""

    product_name: str
    total_amount: float
    order_count: int
    percentage: float


class DashboardStats(BaseModel):
    """Dashboard统计数据"""

    total_amount: float
    total_orders: int
    total_products: int
    this_month_amount: float
    this_month_growth: float  # 环比增长率
    payment_distribution: Dict[str, PaymentDistribution]
    monthly_trend: List[MonthlyTrend]
    top_products: List[TopProduct]

File: frontend/src/services/statistics.ts
/**
 * Statistics API Service
 */
import api from './api';

export interface PaymentDistribution {
  count: number;
  amount: number;
}

export interface MonthlyTrend {
  month: string;
  amount: number;
  count: number;
}

export interface TopProduct {
  product_name: string;
  total_amount: number;
  order_count: number;
  percentage: number;
}

export interface DashboardStats {
  total_amount: number;
  total_orders: number;
  total_products: number;
  this_month_amount: number;
  this_month_growth: number;
  payment_distribution: Record<string, PaymentDistribution>;
  monthly_trend: MonthlyTrend[];
  top_products: TopProduct[];
}

export interface PaymentDueOrder {
  order_id: number;
  order_no: string;
  product_name: string;
  purchase_amount: number;
  order_date: string;
  due_date: string;
  days_remaining: number;
  status: 'overdue' | 'warning' | 'normal';
}

export interface DateRangeStats {
  start_date: string;
  end_date: string;
  total_amount: number;
  total_orders: number;
  avg_order_amount: number;
}

// Get dashboard statistics
export const getDashboardStats = (): Promise<DashboardStats> => {
  return api.get('/statistics/dashboard');
};

// Get monthly trend
export const getMonthlyTrend = (months: number = 6): Promise<MonthlyTrend[]> => {
  return api.get('/statistics/monthly-trend', { params: { months } });
};

// Get payment due orders
export const getPaymentDueOrders = (days_threshold: number = 7): Promise<PaymentDueOrder[]> => {
  return api.get('/statistics/payment-due', { params: { days_threshold } });
};

// Get date range statistics
export const getDateRangeStats = (start_date: string, end_date: string): Promise<DateRangeStats> => {
  return api.get('/statistics/date-range', { params: { start_date, end_date } });
};
